src/status_im2/contexts/wallet/home/view.cljs

```diff
@@ -46,12 +46,19 @@
    {:id :collectibles :label (i18n/label :t/collectibles) :accessibility-label :collectibles-tab}
    {:id :activity :label (i18n/label :t/activity) :accessibility-label :activity-tab}])
 
-(defn view
+(defn f-view
   []
   (let [top          (safe-area/get-top)
         selected-tab (reagent/atom (:id (first tabs-data)))]
     (fn []
-      (let [networks (rf/sub [:wallet/network-details])]
+      (let [networks         (rf/sub [:wallet/network-details])
+            wallet-addresses (rf/sub [:wallet/all-addresses
+                                      networks (rf/sub [:wallet/network-details])])]
+        (rn/use-effect (fn []
+                         (rf/dispatch [:wallet/request-collectibles
+                                       {:addresses    wallet-addresses
+                                        :offset       0
+                                        :new-request? true}])))
```

Comment:
This one is a little complex.

The first thing, is we don't need it to be a functional component, so we don't need to create a `view` wrapper on the `f-view`.
We can pass from:
```clojure
(defn f-view []
  (let [some 'bindings]
    (fn []
      (let [some-other 'bindings]
        (rn/use-effect (fn do-something-once-at-mount []
                         ;;something
                         ))
        [rn/view
         ;; View code
         ]))))
```
to:
```clojure
(defn view []
  (let [_    (fn do-something-once-at-mount []
               ;;something
               )
        some 'bindings]
    (fn []
      (let [some-other 'bindings]
        [rn/view
         ;; View code
         ]))))
```

Reply:
And the second thing is about directly passing a value from a non-parametrized subscription to an event:

```clojure
(defn view []
  (let [my-non-param-sub (rf/sub [:wallet/my-sub])]
    (fn []
      ,,,
      (rf/dispatch [:my-event-with-params my-non-param-sub])
      ,,,
      [rn/view,,,])))
```

that event (`:my-event-with-params`) has access to the db, so it is able to extract the values directly from `app-db` that is in the `cofx` passed to it.

So that event: `:wallet/request-collectibles` can be rewritten as:
```clojure
(rf/reg-event-fx
 :wallet/request-collectibles
 (fn [{:keys [db]} [{:keys [offset new-request?]}]] ;; Removed address param
   (let [addresses      (->> db :profile/wallet-accounts (map :address)) ;; address from app-db 
         request-params [,,,]]
     {,,,})))
```

Reply:
The third thing is about:
```clojure
... wallet-addresses (rf/sub [:wallet/all-addresses networks (rf/sub [,,,])])...
```
that is  non-parametrized sub:
```clojure
(re-frame/reg-sub
 :wallet/all-addresses
 :<- [:profile/wallet-accounts]
 (fn [accounts]
   (map :address accounts)))
```

So we can call it just like: 
`(rf/sub [:wallet/all-addresses])`

Reply:
So, collecting everything, this could be:
```clojure
(defn view []
  (let [_            (rf/dispatch [:wallet/request-collectibles ;; the event dispatched
                                   {:offset       0
                                    :new-request? true}])
        top          (safe-area/get-top)
        selected-tab (reagent/atom (:id (first tabs-data)))]
    (fn []
      (let [networks (rf/sub [:wallet/network-details])]
        [rn/view
         ,,,
         ]))))
```

WDYT? :) 
Feel free to sent me a message if you'd like to chat about this more deeply :+1:

Reply:
Hey @vkjr !

>:wallet/request-collectibles is a wrapper on underlying call, it can be called with different sets of addresses. On this all-accounts screen we retrieve collectibles for all addresses, but on the screen for single account we will pass only one address

Then maybe we could extract the logic to a separate function and call it in different event handlers, using maybe a better name for them:

```clojure
(defn- request-collectibles [addresses]
  ;; all the logic to get collectibles in re-frame
  )

(rf/reg-event-fx
  :wallet/request-collectibles-for-all-addresses
  (fn [...]
    (let [all-addresses (get-all-addresses db)]
      (request-collectibles all-addresses))))


 (rf/reg-event-fx
  :wallet/request-collectibles-for-address
  (fn [ {...} [_ { ... address ... } ]]
      (request-collectibles [address])))
```

So we will not be subscribing in view and just passing the value to an event that can extract that from app-db.

> logic of getting list of all addresses ((map :address accounts)) shouldn't exist in two+ places - subscription and event (or quite possible more events later)

Maybe then we could just extract it to a separate function  (e.g. `get-all-adresses`) and just call it in different event handlers or subs.

> when list of addresses changed for any reason, collectibles should be re-requested and UI should be re-rendered, so we need to rely on (rf/sub [:wallet/all-addresses]) subscription

If we want that, then I think the current implementation is not reflecting that behavior (maybe this explains almost all of the previous mentioned comments).

We should place the `all-addresses` value in the vector of `use-effect`:

```clojure
(let [all-addresses (rf/sub [:wallet/all-addresses])]
(rn/use-effect 
  (fn [] ...)
  [all-addresses])) ;; <- so this effect won't just run at mount, also when `all-addresses` change.
```

Reply:
@ulisesmac,
>We should place the all-addresses value in the vector of use-effect:
That was a good point.
I tried to do this after your comment and ran into a problem - when I specify a collection as a dependency, the hook relaunches every time even if the content of the collection didn't change (and collectibles fetch unnecessarily). I believe it is related to the fact that we pass `cljs->js` converted collections to js implementation of useEffect.

So for now, for the sake of moving this particular PR I did as you suggested:
- made the view non-functional
- put the addresses retrieved inside the event

In the next PR, when we will need to display collectibles for one particular account, I will update the implementation to retrieve from arbitrary addresses, but it is not really needed right now.